### DISCUSSION TITLE
unable to open MATE from command line

### DISCUSSION THREAD

**Discussion by ToddAndMargo** [mate-desktop/mate-desktop#628] [Q&A] [answered]
Hi All,

I am unable to open MATE from the command line (mate-sesson).  I am also not able to open xfce or run xinit from the command line.  

With lightdm greeter, it takes my password, flashes a black screen with a mouse pointer, then goes black to the greeter screen.

But startx does open xfce.   (I have not figured out how to change that to MATE.)

What am I doing wrong?

-T

```
$ mate-session
** (mate-session:8363): WARNING **: 20:26:30.590: Cannot open display:

```

```
$ startxfce4
/usr/bin/startxfce4: Starting X server

(EE) 
Fatal server error:
(EE) Server is already active for display 0
	If this server is no longer running, remove /tmp/.X0-lock
	and start again.
(EE) 
(EE) 
Please consult the The X.Org Foundation support 
	 at http://wiki.x.org
 for help. 
(EE) 
No protocol specified
xinit: giving up
xinit: unable to connect to X server: Resource temporarily unavailable
xinit: server error
```

**Comment by lukefromdc** [comment]
In order to run any x session from the command line, you must first stop any other
instances of X, including lightdm. On systemd-using distros
sudo systemctl stop lightdm
should do it, then you can run x sessions from a TTY

**Reply by ToddAndMargo** [reply]
closed lightdm with "# systemctl stop lightdm".  No symptom change running "$ mate-session" : "Cannot open display:"  startxfce4 no symptom change either.

Note: that this system was working perfectly with lightdm until I did a major "# dnf upgrade".

**Reply by ToddAndMargo** [reply]
I just noticed the following error message come up after attempting to run MATE from lightdm:

This OS version (Fedora Linux 38 [the kernel is 38, the rest is 39] (thirty eight) dracut-059.4.fc38 [rpm -qa dract only shows 39] (init ramfx) is past its end f support d date.

Failed to start systemd-systemd-sysusers.service -- create system user

**Comment by lukefromdc** [comment]
I don't know anything about those errors, but it looks like you may have gotten a bad update
related to systemd or lightdm. We don't maintain either of those at MATE and can do nothing
about this ourselves

**Reply by ToddAndMargo** [reply]
Thank you for the helkp anyway.   Do you happen to know if lightdm has a group and/or forum?  I can't find one.

**Comment by lukefromdc** [comment]
That I do not know, I've never worked on lightdm

**Comment by ToddAndMargo** [comment, marked as answer]
Cancel the request.  It is an issue with lightdm.  I replaced it with lxdm and problems solved.

I am stuck in the "endless loop" problem that is plaguing lightdm users.  I have searched my ass off an everyone
with the issue can not find a cure.

So I disabled lightdm and installed lxdm.  It ain't as pretty BUT IT WORKS!!!!

I even have autologin configured.  And it lets me switch back and forth between Xfce and MATE.

My keeper on lxdm:

```
lxdm auto login:

# vi /etc/lxdm/lxdm.conf

[base]
## uncomment and set autologin username to enable autologin
# autologin=todd
autologin=todd


## default session or desktop used when no systemwide config
# session=/usr/bin/startlxde
session=/usr/bin/startxfce4
```
